File: src/main/java/com/efedotov/vaultly/dto/auth/RecoverRequest.java
package com.efedotov.vaultly.dto.auth;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class RecoverRequest {
    @NotBlank
    private String publicKey;
    @NotBlank
    private String challenge;
    @NotBlank
    private String signature;
}

File: lib/server/model/auth/recover_request.dart
class RecoverRequest {
  final String publicKey;
  final String challenge;
  final String signature;

  RecoverRequest({
    required this.publicKey,
    required this.challenge,
    required this.signature,
  });

  factory RecoverRequest.fromJson(Map<String, dynamic> json) {
    return RecoverRequest(
      publicKey: json['publicKey'] as String,
      challenge: json['challenge'] as String,
      signature: json['signature'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
    'publicKey': publicKey,
    'challenge': challenge,
    'signature': signature,
  };
}
